contracts/JBERC20PaymentTerminal3_1_2.sol

```diff
@@ -56,8 +52,7 @@ contract JBERC20PaymentTerminal3_1_2 is JBPayoutRedemptionPaymentTerminal3_1_2 {
     JBPayoutRedemptionPaymentTerminal3_1_2(
       address(_token),
       _token.decimals(),
-      _currency,
-      _baseWeightCurrency,
+      uint256(uint24(uint160(address(_token)))), // last 24 bits used for currency.
```

Comment:
shouldn't we have some gatekeeper here, in case we're extremly unlucky (eg a param to override this currency, in case we end up with 2 same ones for different tokens)?
```
_currencyOverriden != 0 ? _currencyOverriden : uint256(uint24(uint160(address(_token))))
```